components/script/dom/htmlelement.rs

```diff
@@ -151,7 +151,12 @@ impl HTMLElementMethods for HTMLElement {
     // https://html.spec.whatwg.org/multipage/#handler-onload
     fn GetOnload(&self) -> Option<Rc<EventHandlerNonNull>> {
         if self.is_body_or_frameset() {
-            window_from_node(self).GetOnload()
+            let document = document_from_node(self);
+            if document.has_browsing_context() {
+                document.window().GetOnload()
+            } else {
+                None
```

Comment:
> If an event handler IDL attribute exposes an event handler of an object that doesn't exist, it must always return null on getting and must do nothing on setting.

Here, you did not check whether an event handler exists when a browsing context doesn't exist.